data/trie/leafNode_test.go

```diff
@@ -286,23 +310,25 @@ func TestLeafNode_getNext(t *testing.T) {
 
 func TestLeafNode_getNextWrongKey(t *testing.T) {
 	t.Parallel()
+
 	db := mock.NewMemDbMock()
 	ln := getLn()
 	marsh, _ := getTestMarshAndHasher()
-	key := []byte{2, 100, 111, 103}
+	wrongKey := []byte{2, 100, 111, 103}
 
-	node, key, err := ln.getNext(key, db, marsh)
+	node, key, err := ln.getNext(wrongKey, db, marsh)
 	assert.Nil(t, node)
 	assert.Nil(t, key)
 	assert.Equal(t, ErrNodeNotFound, err)
 }
 
 func TestLeafNode_getNextNilNode(t *testing.T) {
 	t.Parallel()
+
 	db := mock.NewMemDbMock()
 	var ln *leafNode
 	marsh, _ := getTestMarshAndHasher()
-	key := []byte{2, 100, 111, 103}
+	key := []byte("dog")
```

Comment:
you missed something here ? _dog_ as I see is _100,111,103_ but you missed the byte of 2

Reply:
The key does not matter, because the leafNode is nil, and we test that if we call getNext() on a nil node, it returns an error. So the key does not matter. Replaced []byte{2, 100, 111, 103} with []byte("dog") because it is easier to read.